### PR TITLE
C++ Tracing Docs

### PR DESCRIPTION
**Pull request by willgittoes-dd**
Add examples of how to accomplish tracing, and document it in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,17 +4,270 @@
...
+Include `<datadog/opentracing.h>` and create the tracer:
+
+```cpp
+// tracer_example.cpp
+#include <datadog/opentracing.h>
+#include <iostream>
+#include <string>
+
+int main(int argc, char* argv[]) {
+  datadog::opentracing::TracerOptions tracer_options{"dd-agent", 8126, "compiled-in example"};
```
Comment (cgilmour) [inline README.md]: This should probably be `localhost`

File: README.md [hunk 2]
```diff
@@ -4,17 +4,270 @@
...
+      "/usr/local/lib/libdd_opentracing_plugin.so", error_message);
+  if (!handle_maybe) {
+    std::cerr << "Failed to load tracer library " << error_message << "\n";
+    return false;
+  }
+
+  // Read in the tracer's configuration.
+  std::string tracer_config = R"({
+      "service": "dynamic-load example",
+      "agent_host": "dd-agent",
```
Comment (cgilmour) [inline README.md]: Also this.
It'll be fine in docker-compose with a container named `dd-agent`, but that's not necessarily how users will run their apps.

File: README.md [hunk 3]
```diff
@@ -4,17 +4,270 @@
...
+
+  tracer->Close();
+  return 0;
+}
+```
+
+Just link against libopentracing (making sure that libopentracing.so is in your LD_LIBRARY_PATH):
+
+```bash
+g++ -o tracer_example tracer_example.cpp -lopentracing
```
Comment (cgilmour) [inline README.md]: Needs `-ldl` too.
Should it be named differently from the other example?

File: README.md [hunk 4]
```diff
@@ -4,17 +4,270 @@
...
+##### Debugging
+
+The release binary libraries are all compiled with debug symbols added to the optimized release. It is possible to use gdb or lldb to debug the library and to read core dumps. If you are building the library from source, pass the argument `-DCMAKE_BUILD_TYPE=RelWithDebInfo` to cmake to compile an optimized build with debug symbols.
+
+```bash
+cd .build
+cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo ..
+make
+make install
+```
```
Comment (cgilmour) [inline README.md]: Can the `building` section at the end be dropped?
Maybe it belongs in a developer guide instead, or a separate section for actually developing/testing/contributing to this codebase.

File: README.md (hunk 1 above)
Comment (willgittoes-dd) [reply README.md]: I made it the same code as what is used in the example, which runs on Docker.

File: README.md (hunk 2 above)
Comment (willgittoes-dd) [reply README.md]: I wasn't sure which was worse: very slightly different code in the example and the README, or a README code that needs a small change to work on local machines!

File: README.md (hunk 3 above)
Comment (willgittoes-dd) [reply README.md]: Works fine without libdl! I thought dlopen was in libc with just extra stuff in libdl?

**Comment by palazzem** [comment]
Thanks @cgilmour !

**Pull request merged**